python/src/uagents/wallet_messaging.py

```diff
@@ -0,0 +1,82 @@
+import asyncio
+import base64
+import functools
+import logging
+from typing import List, Optional
+
+from babble import Client, Identity as BabbleIdentity
+from babble.client import Message as WalletMessage
+from cosmpy.aerial.wallet import LocalWallet
+from requests import HTTPError, JSONDecodeError
+
+from uagents.config import (
+    WALLET_MESSAGING_POLL_INTERVAL_SECONDS,
+    get_logger,
+)
+from uagents.context import Context, WalletMessageCallback
+from uagents.crypto import Identity
+
+
+class WalletMessagingClient:
+    def __init__(
+        self,
+        identity: Identity,
+        wallet: LocalWallet,
+        chain_id: str,
+        logger: Optional[logging.Logger] = None,
+    ):
+        delegate_pubkey = identity.pub_key
+        delegate_pubkey_b64 = base64.b64encode(bytes.fromhex(delegate_pubkey)).decode()
+        public_key = base64.b64decode(wallet.public_key().public_key).hex()
+        signed_bytes, signature = identity.sign_arbitrary(public_key.encode())
+        self._client = Client(
+            identity.address,
+            delegate_pubkey_b64,
+            signature,
+            signed_bytes,
+            BabbleIdentity(wallet.signer().private_key_bytes),
+            chain_id,
+        )
+        self._poll_interval = WALLET_MESSAGING_POLL_INTERVAL_SECONDS
+        self._logger = logger or get_logger("wallet_messaging")
+        self._message_queue = asyncio.Queue()
+        self._message_handlers: List[WalletMessageCallback] = []
+
+    def on_message(
+        self,
+    ):
+        def decorator_on_message(func: WalletMessageCallback):
+            @functools.wraps(func)
+            def handler(*args, **kwargs):
+                return func(*args, **kwargs)
+
+            self._message_handlers.append(func)
+
+            return handler
+
+        return decorator_on_message
+
+    async def send(self, destination: str, msg: WalletMessage, msg_type: int = 1):
+        self._client.send(destination, msg, msg_type)
```

Comment:
The client `.send()` method requires a string but the WalletMessage is an object composed of multiple fields.
Did you mean `msg.text`? Based on the example I would assume we change the `msg` attribute for the `WalletMessagingClient` to `str` as well.

Reply:
Good catch, actually the type hint in the function header is wrong and `WalletMessage` is not needed. Fixed.